### DISCUSSION TITLE
Ability to block a tab from accidental clicks

### DISCUSSION THREAD

**Discussion by trollqaa** [ghostty-org/ghostty#3575] [Feature Requests, Ideas] [answered]
Hello.
I would like to see in your wonderful terminal a function to block keyboard input by pressing a shortcut key, as is implemented in the windows terminal

This is very convenient when an important task is running and you don’t want to accidentally interrupt it, but you still need to monitor its progress (this is an answer in advance to the fact that you can open another tab/window)

Sorry for my english (i use GTranslate)

**Comment by mitchellh** [comment]
Interesting idea.

> I would like to see in your wonderful terminal a function to block keyboard input by pressing a shortcut key, as is implemented in the windows terminal

Can you please direct me to the docs for this?

**Reply by trollqaa** [reply]
Hello.
https://learn.microsoft.com/en-us/windows/terminal/panes#marking-a-pane-as-read-only

And on the Windows terminal side, in the hotkey settings, there is an item "Enable and disable read-only mode for a pane"

**Comment by mitchellh** [comment]
Okay, this is interesting. Based on the WinTerm docs, it seems that this:

1. Blocks all input (mouse and keyboard), though I assume things like selection, copy/paste, scrolling still work
2. Warns if you attempt to close the terminal window

If we wanted to do this and block _all_ input including selection, it'd be pretty trivial (can do purely in apprts without libghostty even knowing). But if we want to allow selection, scrolling and so on, then this has to be a core surface concern. 

I think it's a pretty interesting use case. I'm supportive of adding this. I'm going to let this discussion sit for a second to solicit any other feedback but I'm inclined to accept it.

**Reply by pluiedev** [reply]
I think this would be nice to add as well, especially when you just want to monitor a long-running process (e.g. in a presentation) while avoiding doing anything that could disrupt it. It sounds like we can just implement this by intercepting and cancelling all key & mouse events that get passed onwards to the terminal I/O, which shouldn't be hard at all

**Comment by mitchellh** [comment, marked as answer]
#8432 accepted.

**Comment by matthew-hre** [comment]
Started doing some work on this (#9130), but wanted to jump back here to ask if there were any ideas around a visual indicator to show that the surface is read-only? Maybe throwing a lock in the titlebar or a border of some kind?
